MovieGallery/Pages/Search.cshtml.cs

```diff
@@ -44,28 +46,36 @@ public void OnGet(string SearchbyYear)
             {
                 Query = SearchbyYear;
                 long inputYear = Convert.ToInt64(Query);
-
-                var result = MoviesJson.Items.Where(x => x.Year == inputYear).ToList();
-                MoviesResult = new Movies()
+                if (inputYear > 1000 && inputYear <= Convert.ToInt64(currentYear))
```

Comment:
Condition added to check if the input is in correct format and a future year has not been entered.